File: Coding_ObjectiveC/Node.m
//
//  Node.m
//  Coding_ObjectiveC
//
//  Created by Alexander Kobylinskyy on 12/21/13.
//  Copyright (c) 2013 Alex. All rights reserved.
//

#import "Node.h"

@implementation Node

#pragma mark - Initialization

- (id)initWithData:(NSInteger)data andNextNode:(Node *)next {
	self = [super init];
	if (self) {
		_data = data;
		_next = next;
	}
	return self;
}

- (id)initWithData:(NSInteger)data {
	self = [self initWithData:data andNextNode:nil];
	if (self) {
	}
	return self;
}

- (id)initWithNextNode:(Node *)next {
	self = [self initWithData:0 andNextNode:next];
	if (self) {
	}
	return self;
}

#pragma mark - Description

- (NSString *)description {
	NSString *description = [NSString stringWithFormat:@"%li", (long)_data];
	return description;
}

- (NSString *)debugDescription {
	NSString *description = [NSString stringWithFormat:@"data:%li next:%@", (long)_data, _next ? @"is present" : @"is not present"];
	return description;
}

@end

File: Coding_ObjectiveC/Node.swift
//
//  Node.swift
//  Coding_ObjectiveC
//
//  Created by Alexander Kobylinskyy on 8/24/14.
//  Copyright (c) 2014 Alex. All rights reserved.
//

import Foundation

class Node<T>: Printable {
	typealias Element = T
	var data: T!
	var nextNode: Node<T>? = nil
	
	init(data: T, nextNode: Node<T>? = nil) {
		self.data = data
		self.nextNode = nextNode
	}
	
	var description: String {
		return "Node:\(data) next:\(nextNode)"
	}
}
